Index: kdeui/util/kcrash.cpp
===================================================================
--- kdeui/util/kcrash.cpp
+++ kdeui/util/kcrash.cpp	2010-11-29 18:22:51.000000000 +0100
@@ -200,19 +200,7 @@
 
 void KCrash::setDrKonqiEnabled(bool enabled)
 {
-    s_launchDrKonqi = enabled;
-    if (s_launchDrKonqi && !s_drkonqiPath) {
-        s_drkonqiPath = qstrdup(QFile::encodeName(KStandardDirs::findExe("drkonqi")).constData());
-        if (!s_drkonqiPath) {
-            kError() << "Could not find drkonqi";
-            s_launchDrKonqi = false;
-        }
-    }
-
-    //we need at least the default crash handler to launch drkonqi
-    if (s_launchDrKonqi && !s_crashHandler) {
-        setCrashHandler(defaultCrashHandler);
-    }
+    s_launchDrKonqi = false;
 }
 
 bool KCrash::isDrKonqiEnabled()
