diff -urN kdelibs-4.13.2/cmake/modules/CMakeLists.txt kdelibs-4.13.2-patched/cmake/modules/CMakeLists.txt
--- kdelibs-4.13.2/cmake/modules/CMakeLists.txt	2014-06-11 18:26:45.446788308 +1100
+++ kdelibs-4.13.2-patched/cmake/modules/CMakeLists.txt	2014-06-11 18:27:06.573139954 +1100
@@ -171,8 +171,3 @@
 #   file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/remove_files.cmake "exec_program( ${CMAKE_COMMAND} ARGS -E remove ${_current_FILE} OUTPUT_VARIABLE _dummy)\n" )
 #endforeach ( _current_FILE)
 
-if(NOT CMAKE_VERSION VERSION_GREATER 2.8.12)
-  # create and install a man page for the kdelibs cmake modules
-  add_custom_target(KDECMakeModulesManPage ALL COMMAND ${CMAKE_COMMAND} -DCMAKE_MODULE_PATH=${CMAKE_CURRENT_SOURCE_DIR} --help-custom-modules ${CMAKE_CURRENT_BINARY_DIR}/kdecmake.1)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/kdecmake.1 DESTINATION "${MAN_INSTALL_DIR}/man1")
-endif()
