diff -Naur kdelibs-4.10.0/khtml/imload/decoders/gifloader.cpp kdelibs-4.10.0-giflib5/khtml/imload/decoders/gifloader.cpp
--- kdelibs-4.10.0/khtml/imload/decoders/gifloader.cpp	2013-01-23 21:44:22.000000000 +0000
+++ kdelibs-4.10.0-giflib5/khtml/imload/decoders/gifloader.cpp	2013-02-11 14:47:32.856000000 +0000
@@ -300,7 +300,11 @@
     }
     
     
+#if GIFLIB_MAJOR >= 5
+    static unsigned int decode16Bit(unsigned char* signedLoc)
+#else
     static unsigned int decode16Bit(char* signedLoc)
+#endif
     {
         unsigned char* loc = reinterpret_cast<unsigned char*>(signedLoc);
     
@@ -348,7 +352,11 @@
     virtual int processEOF()
     {
         //Feed the buffered data to libUnGif
+#if GIFLIB_MAJOR >= 5
+        GifFileType* file = DGifOpen(this, gifReaderBridge, NULL);
+#else
         GifFileType* file = DGifOpen(this, gifReaderBridge);
+#endif
         
         if (!file)
             return Error;
