diff -urN kdelibs-4.8.95/kfile/kfileplaceeditdialog.cpp kdelibs-4.8.95-patched/kfile/kfileplaceeditdialog.cpp
--- kdelibs-4.8.95/kfile/kfileplaceeditdialog.cpp	2012-05-23 10:44:07.000000000 +1100
+++ kdelibs-4.8.95-patched/kfile/kfileplaceeditdialog.cpp	2012-07-09 20:19:47.214731283 +1100
@@ -107,7 +107,7 @@
                                  "the location's URL.</qt>");
     m_labelEdit = new KLineEdit(wdg);
     layout->addRow(i18n("L&abel:"), m_labelEdit);
-    m_labelEdit->setText(label);
+    m_labelEdit->setText(i18n(label.toUtf8()));
     m_labelEdit->setClickMessage(i18n("Enter descriptive label here"));
     m_labelEdit->setWhatsThis(whatsThisText);
     layout->labelForField(m_labelEdit)->setWhatsThis(whatsThisText);
diff -urN kdelibs-4.8.95/kfile/kfileplacesmodel.cpp kdelibs-4.8.95-patched/kfile/kfileplacesmodel.cpp
--- kdelibs-4.8.95/kfile/kfileplacesmodel.cpp	2012-06-20 08:45:52.000000000 +1100
+++ kdelibs-4.8.95-patched/kfile/kfileplacesmodel.cpp	2012-07-09 20:25:01.373750146 +1100
@@ -17,8 +17,10 @@
     Boston, MA 02110-1301, USA.
 
 */
+
 #include "kfileplacesmodel.h"
 #include "kfileplacesitem_p.h"
+#include "kglobalsettings.h"
 
 #ifdef _WIN32_WCE
 #include "Windows.h"
@@ -116,6 +118,22 @@
         KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
                                               "Home", I18N_NOOP2("KFile System Bookmarks", "Home"),
                                               KUrl(KUser().homeDir()), "user-home");
+        // We add Documents, Music, Downloads, Pictures, Videos
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Documents", I18N_NOOP2("KFile System Bookmarks", "Documents"),
+                                              KUrl(KGlobalSettings::documentPath()), "folder-documents");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Music", I18N_NOOP2("KFile System Bookmarks", "Music"),
+                                              KUrl(KGlobalSettings::musicPath()), "folder-sound");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Downloads", I18N_NOOP2("KFile System Bookmarks", "Downloads"),
+                                              KUrl(KGlobalSettings::downloadPath()), "folder-downloads");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Videos", I18N_NOOP2("KFile System Bookmarks", "Videos"),
+                                              KUrl(KGlobalSettings::videosPath()), "folder-video");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Pictures", I18N_NOOP2("KFile System Bookmarks", "Pictures"),
+                                              KUrl(KGlobalSettings::picturesPath()), "folder-image");
         KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
                                               "Network", I18N_NOOP2("KFile System Bookmarks", "Network"),
                                               KUrl("remote:/"), "network-workgroup");
