diff -urN kdelibs-4.12.4/khtml/imload/decoders/gifloader.cpp kdelibs-4.12.4-patched/khtml/imload/decoders/gifloader.cpp
--- kdelibs-4.12.4/khtml/imload/decoders/gifloader.cpp	2014-03-28 07:06:12.000000000 +1100
+++ kdelibs-4.12.4-patched/khtml/imload/decoders/gifloader.cpp	2014-05-28 23:42:14.611422462 +1100
@@ -369,7 +369,11 @@
         
         if (DGifSlurp(file) == GIF_ERROR)
         {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+            DGifCloseFile(file, NULL);
+#else
             DGifCloseFile(file);
+#endif
             return Error;
         }
         
@@ -378,7 +382,11 @@
         if (file->ImageCount > 1) {
             // Verify it..
             if (!ImageManager::isAcceptableSize(file->SWidth, file->SHeight)) {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+                DGifCloseFile(file, NULL);
+#else
                 DGifCloseFile(file);
+#endif
                 return Error;
             }
             notifyImageInfo(file->SWidth, file->SHeight);
@@ -390,7 +398,11 @@
             //Extract colormap, geometry, so that we can create the frame
             SavedImage* curFrame = &file->SavedImages[frame];
             if (!ImageManager::isAcceptableSize(curFrame->ImageDesc.Width, curFrame->ImageDesc.Height)) {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+                DGifCloseFile(file, NULL);
+#else
                 DGifCloseFile(file);
+#endif
                 return Error;
             }
         }
@@ -576,7 +588,11 @@
             frame0->animProvider = new GIFAnimProvider(frame0, image, frameProps, bgColor);
         }
         
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+        DGifCloseFile(file, NULL);
+#else
         DGifCloseFile(file);
+#endif
 
         return Done;
     }
