diff -urN kdelibs-4.12.4/kio/kio/global.h kdelibs-4.12.4-patched/kio/kio/global.h
--- kdelibs-4.12.4/kio/kio/global.h	2014-03-28 07:06:12.000000000 +1100
+++ kdelibs-4.12.4-patched/kio/kio/global.h	2014-04-18 00:39:52.846015369 +1100
@@ -26,7 +26,7 @@
 #include <QtCore/QMap>
 #include <QtCore/QList>
 #include <QtCore/QVariant>
-#include <QFile>  // for QFile::Permissions
+#include <QtCore/QFile>  // for QFile::Permissions
 
 #include <kiconloader.h>
 #include <QtGui/QPixmap> // for pixmapForUrl
