diff --git a/cmake/modules/CMakeLists.txt b/cmake/modules/CMakeLists.txt
index 94f8da0..4cb7293 100644
--- a/cmake/modules/CMakeLists.txt
+++ b/cmake/modules/CMakeLists.txt
@@ -171,7 +171,7 @@ install(FILES ${cmakeFiles}
 #   file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/remove_files.cmake "exec_program( ${CMAKE_COMMAND} ARGS -E remove ${_current_FILE} OUTPUT_VARIABLE _dummy)\n" )
 #endforeach ( _current_FILE)
 
-if(NOT CMAKE_VERSION VERSION_GREATER 2.8.12)
+if(NOT CMAKE_VERSION VERSION_GREATER 2.8.12.2)
   # create and install a man page for the kdelibs cmake modules
   add_custom_target(KDECMakeModulesManPage ALL COMMAND ${CMAKE_COMMAND} -DCMAKE_MODULE_PATH=${CMAKE_CURRENT_SOURCE_DIR} --help-custom-modules ${CMAKE_CURRENT_BINARY_DIR}/kdecmake.1)
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/kdecmake.1 DESTINATION "${MAN_INSTALL_DIR}/man1")
